startreedata/thirdeye
# [kubernetes] sample dataset ingestion

#### Issue(s)

https://cortexdata.atlassian.net/browse/TE-1094

#### Description

To enable sample data creation along with the ThirdEye installation we need to provide a helper job that uploads the sample datasets to the user provided pinot instance so that this data can be used later to create sample alerts for first time users.

This PR introduces :

1. A k8s `Job` that checks pinot connectivity and then pushes sample data to it.
2. A k8s `ConfigMap` that holds the script which the job runs to upload data.

values.yaml properties : 

- `dataloader.enabled` : flag to enable/disable the sample data ingestion job. Default - `false` 
- `dataloader.pinot.controllerProtocol` : pinot controller protocol http/https. Default - `http`
- `dataloader.pinot.controllerHost` : pinot controller host dns. Default - `pinot-pinot-controller-headless`
- `dataloader.pinot.controllerPort` : pinot controller port. Default - `9000`
- `dataloader.pinot.commandOptions` : any extra command line options to pass to the pinot-admin.sh tool such as if auth is enabled on pinot then pass `-authToken 'Basic <auth token>'`. Default - none
- `dataloader.sampleDatasets` : it expects an array of sample dataset base urls. Please follow the sample dataset directory structure as followed by the existing datasets in examples directory of this repo. The base url should be such that files must be downloadable over http, for example `https://raw.githubusercontent.com/startreedata/thirdeye/master/examples/order_events`

#### How to test

1. Run a pinot instance in local
2. Pass these dataloader values to the values.yaml file
```
dataloader:
  enabled: true
  pinot:
    controllerHost: <machine ip such as 192.168.1.2>
  sampleDatasets:
    - https://raw.githubusercontent.com/startreedata/thirdeye/master/examples/order_events
    - https://raw.githubusercontent.com/startreedata/thirdeye/master/examples/pageviews
```
3. Install the helm charts
4. Once the pod for the dataloader job goes to completion check pinot for the tables
